include/envoy/network/address.h

```diff
@@ -128,11 +129,11 @@ class Instance {
   /**
    * Bind a socket to this address. The socket should have been created with a call to socket() on
    * an Instance of the same address family.
-   * @param fd supplies the platform socket handle.
+   * @param io_handle supplies the platform I/O handle.
    * @return a Api::SysCallIntResult with rc_ = 0 for success and rc_ = -1 for failure. If the call
    *   is successful, errno_ shouldn't be used.
    */
-  virtual Api::SysCallIntResult bind(int fd) const PURE;
+  virtual Api::SysCallIntResult bind(IoHandle& io_handle) const PURE;
```

Comment:
Just to make sure we are all on the same page, the end goal here is to eventually reverse this so we have `io_handle.bind(...)` right?

Reply:
Correct. Once we decide on the proper form of this first PR, then the next round would be to move all the methods into the appropriate type of iohandle.